### PR TITLE
`azurerm_log_analytics_workspace`: allow changing `sku` between `CapacityReservation` and `PerGB2018`

### LINKED ISSUE #12177: azurerm_log_analytics_workspace: on change of  "sku", terraform incorrectly wants to rebuild workspace
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.13.7
+ provider registry.terraform.io/hashicorp/azurerm v2.63.0

### Affected Resource(s)

* azurerm_log_analytics_workspace

### Terraform Configuration Files

```hcl
resource azurerm_log_analytics_workspace log_analytics_workspace {
  name                = format("%s-%s-%s-%s", var.resource_name_prefix.log_analytics_workspace, var.project_name, "02", var.stage)
  location            = module.resource_group.location
  resource_group_name = module.resource_group.name
  # sku                 = "PerGB2018"
  sku = "CapacityReservation"
  reservation_capcity_in_gb_per_day = 100
  daily_quota_gb    = local.daily_quota_gb
  retention_in_days = var.retention_period
  tags = tags
}
```

### Expected Behaviour

When changing the sku from "PerGB2018" to "CapacityReservation", the workspace should just have been updated, not destroyed. This change is easily doable in the portal or REST API without the workspace getting re-created.

### Actual Behaviour

Terraform wants to delete the workspace due to the sku change:
```
  # azurerm_log_analytics_workspace.log_analytics_workspace must be replaced
-/+ resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
        daily_quota_gb             = 200
      ~ id                         = "/subscriptions/<edited>/resourceGroups/rg-monitoring-prd/providers/Microsoft.OperationalInsights/workspaces/<edited>" -> (known after apply)
        internet_ingestion_enabled = true
        internet_query_enabled     = true
        location                   = "westeurope"
        name                       = "<edited>"
      + portal_url                 = (known after apply)
      ~ primary_shared_key         = (sensitive value)
        resource_group_name        = "rg-monitoring-prd"
        retention_in_days          = 365
      ~ secondary_shared_key       = (sensitive value)
      ~ sku                        = "pergb2018" -> "CapacityReservation" # forces replacement
        tags                       = {
            "ProjectName"  = "monitoring"
        }
      ~ workspace_id               = "<edited>" -> (known after apply)
    }

```

### Steps to Reproduce

1. change the sku, run `terraform apply`
2. terraform says it wants to replace the workspace

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #12177

changing between `CapacityReservation` and `PerGB2018` does not force new resource to be created, but it will lead to a 31-days commitment period during which it could not be changed to lower pricing tier.  https://learn.microsoft.com/en-us/azure/azure-monitor/logs/cost-logs#commitment-tiers)

test
---
```
❯ tftest loganalytics TestAccLogAnalyticsWorkspace    
=== RUN   TestAccLogAnalyticsWorkspace_basic
=== PAUSE TestAccLogAnalyticsWorkspace_basic
=== RUN   TestAccLogAnalyticsWorkspace_requiresImport
=== PAUSE TestAccLogAnalyticsWorkspace_requiresImport
=== RUN   TestAccLogAnalyticsWorkspace_complete
=== PAUSE TestAccLogAnalyticsWorkspace_complete
=== RUN   TestAccLogAnalyticsWorkspace_freeTier
    log_analytics_workspace_resource_test.go:95: `TestAccLogAnalyticsWorkspace_freeTier` due to subscription pricing tier configuration (e.g. Pricing tier doesn't match the subscription's billing model.)
--- SKIP: TestAccLogAnalyticsWorkspace_freeTier (0.00s)
=== RUN   TestAccLogAnalyticsWorkspace_withDefaultSku
=== PAUSE TestAccLogAnalyticsWorkspace_withDefaultSku
=== RUN   TestAccLogAnalyticsWorkspace_withVolumeCap
=== PAUSE TestAccLogAnalyticsWorkspace_withVolumeCap
=== RUN   TestAccLogAnalyticsWorkspace_removeVolumeCap
=== PAUSE TestAccLogAnalyticsWorkspace_removeVolumeCap
=== RUN   TestAccLogAnalyticsWorkspace_withInternetIngestionEnabled
=== PAUSE TestAccLogAnalyticsWorkspace_withInternetIngestionEnabled
=== RUN   TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== PAUSE TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== RUN   TestAccLogAnalyticsWorkspace_withCapacityReservation
=== PAUSE TestAccLogAnalyticsWorkspace_withCapacityReservation
=== RUN   TestAccLogAnalyticsWorkspace_negativeOne
=== PAUSE TestAccLogAnalyticsWorkspace_negativeOne
=== RUN   TestAccLogAnalyticsWorkspace_cmkForQueryForced
=== PAUSE TestAccLogAnalyticsWorkspace_cmkForQueryForced
=== RUN   TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission
=== PAUSE TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission
=== RUN   TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
=== PAUSE TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
=== RUN   TestAccLogAnalyticsWorkspace_updateSku
=== PAUSE TestAccLogAnalyticsWorkspace_updateSku
=== CONT  TestAccLogAnalyticsWorkspace_basic
=== CONT  TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== CONT  TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission
=== CONT  TestAccLogAnalyticsWorkspace_withVolumeCap
=== CONT  TestAccLogAnalyticsWorkspace_negativeOne
=== CONT  TestAccLogAnalyticsWorkspace_requiresImport
=== CONT  TestAccLogAnalyticsWorkspace_withDefaultSku
=== CONT  TestAccLogAnalyticsWorkspace_complete
=== NAME  TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
    testcase.go:113: Error running init: exit status 1
        
        Error: Failed to install provider
        
        Error while installing hashicorp/azuread v2.8.0: read tcp
        198.18.0.1:52300->198.18.0.44:443: read: operation timed out
        
--- PASS: TestAccLogAnalyticsWorkspace_requiresImport (146.76s)
=== CONT  TestAccLogAnalyticsWorkspace_updateSku
=== CONT  TestAccLogAnalyticsWorkspace_withInternetIngestionEnabled
--- FAIL: TestAccLogAnalyticsWorkspace_withInternetQueryEnabled (148.63s)
--- PASS: TestAccLogAnalyticsWorkspace_withDefaultSku (166.74s)
=== CONT  TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
--- PASS: TestAccLogAnalyticsWorkspace_complete (173.74s)
=== CONT  TestAccLogAnalyticsWorkspace_cmkForQueryForced
--- PASS: TestAccLogAnalyticsWorkspace_negativeOne (214.52s)
=== CONT  TestAccLogAnalyticsWorkspace_withCapacityReservation
--- PASS: TestAccLogAnalyticsWorkspace_basic (229.89s)
=== CONT  TestAccLogAnalyticsWorkspace_removeVolumeCap
--- PASS: TestAccLogAnalyticsWorkspace_withVolumeCap (250.41s)
--- PASS: TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission (316.67s)
--- PASS: TestAccLogAnalyticsWorkspace_updateSku (183.77s)
--- PASS: TestAccLogAnalyticsWorkspace_withInternetIngestionEnabled (186.86s)
--- PASS: TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth (177.66s)
--- PASS: TestAccLogAnalyticsWorkspace_cmkForQueryForced (176.09s)
--- PASS: TestAccLogAnalyticsWorkspace_withCapacityReservation (209.43s)
--- PASS: TestAccLogAnalyticsWorkspace_removeVolumeCap (216.31s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics  446.273s
FAIL
```

rerun the failed one
```
❯ tftest loganalytics TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== RUN   TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== PAUSE TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
=== CONT  TestAccLogAnalyticsWorkspace_withInternetQueryEnabled
--- PASS: TestAccLogAnalyticsWorkspace_withInternetQueryEnabled (144.83s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics  144.886s
```

### CODE REVIEW

File: website/docs/r/log_analytics_workspace.html.markdown [hunk 1]
```diff
@@ -45,6 +45,8 @@ The following arguments are supported:
 
 ~> **NOTE:** A new pricing model took effect on `2018-04-03`, which requires the SKU `PerGB2018`. If you're provisioned resources before this date you have the option of remaining with the previous Pricing SKU and using the other SKUs defined above. More information about [the Pricing SKUs is available at the following URI](https://aka.ms/PricingTierWarning).
 
+~> **NOTE:** Changing `sku` forces a new Log Analytics Workspace to be created, except changing between `PerGB2018` and `CapacityReservation`, but changing it to `CapacityReservation` or changing `reservation_capacity_in_gb_per_day` to a higher tier will lead to a 31-days commitment period, during which the sku could not be changed to a lower one. More information about [the commitment perioud](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/cost-logs#commitment-tiers)
```
Comment (manicminer) [inline website/docs/r/log_analytics_workspace.html.markdown:48]:
```suggestion
~> **NOTE:** Changing `sku` forces a new Log Analytics Workspace to be created, except when changing between `PerGB2018` and `CapacityReservation`. However, changing `sku` to `CapacityReservation` or changing `reservation_capacity_in_gb_per_day` to a higher tier will lead to a 31-days commitment period, during which the SKU cannot be changed to a lower one. Please refer to [official documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/cost-logs#commitment-tiers) for further information.
```

**Review by manicminer — CHANGES_REQUESTED** [review]
Thanks @ms-henglu, this is off to a good start, however we'll need to avoid using `d.GetRawConfig()`. If you can update this then we'll circle back for re-review and testing. Thanks!

**Comment by ziyeqf** [comment]
we might need to note `reservation_capacity_in_gb_per_day` is not `computed` yet in changelog?

**Comment by katbyte** [comment]
@ziyeqf - what do you mean by `reservation_capacity_in_gb_per_day is not computed yet` ?

**Comment by ziyeqf** [comment]
> @ziyeqf - what do you mean by `reservation_capacity_in_gb_per_day is not computed yet` ?

oh, I mean it's not `computed` anymore.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by ziyeqf** [comment]
kindly ping for merge :)

**Pull request merged**